src/Container/NotFoundHandlerFactory.php

```diff
@@ -10,13 +10,23 @@
 namespace Zend\Expressive\Container;
 
 use Psr\Container\ContainerInterface;
-use Zend\Expressive\Delegate\NotFoundDelegate;
-use Zend\Expressive\Middleware\NotFoundHandler;
+use Zend\Diactoros\Response;
+use Zend\Expressive\Handler\NotFoundHandler;
+use Zend\Expressive\Template\TemplateRendererInterface;
 
 class NotFoundHandlerFactory
 {
     public function __invoke(ContainerInterface $container) : NotFoundHandler
     {
-        return new NotFoundHandler($container->get(NotFoundDelegate::class));
+        $config   = $container->has('config') ? $container->get('config') : [];
+        $renderer = $container->has(TemplateRendererInterface::class)
+            ? $container->get(TemplateRendererInterface::class)
+            : null;
+        $template = $config['zend-expressive']['error_handler']['template_404']
+            ?? NotFoundHandler::TEMPLATE_DEFAULT;
+        $layout = $config['zend-expressive']['error_handler']['layout']
+            ?? NotFoundHandler::LAYOUT_DEFAULT;
+
+        return new NotFoundHandler(new Response(), $renderer, $template, $layout);
```

Comment:
Should pull `$container->get(ResponseInterface::class)` here?

Reply:
Yes - thanks!